src/cli/login.rs

```diff
@@ -132,42 +144,48 @@ impl<Backend: Interact> Login<Backend> {
         client: reqwest::Client,
         verify_credentials: bool,
         store_password_in_plaintext: bool,
-    ) -> Result<config::User, lib::error::UserFriendly> {
+    ) -> Result<(), lib::error::UserFriendly> {
         let mut user_list: Vec<String> = config.get_users().iter().map(combine_username).collect();
 
-        if user_list.is_empty() {
+        let mut user = if user_list.is_empty() {
             writeln!(
                 console,
                 "no stored api credentials found; add a new user to continue:"
             )
             .unwrap();
-            let user = self.input_user();
+            let mut user = self.input_user();
 
             if verify_credentials {
                 test_request(client, config, &user).await?;
+                user.last_used = Some(chrono::offset::Utc::now());
             }
 
             config.add_user(user.clone(), store_password_in_plaintext)?;
-            Ok(user)
+            user
         } else {
             const ADD_A_USER_OPTION: &str = "add a user";
             user_list.push(ADD_A_USER_OPTION.to_owned());
 
             let selection = self.interact.select("select a user", 0, &user_list);
 
             if user_list[selection] == ADD_A_USER_OPTION {
-                let user = self.input_user();
+                let mut user = self.input_user();
 
                 if verify_credentials {
                     test_request(client, config, &user).await?;
+                    user.last_used = Some(chrono::offset::Utc::now());
                 }
 
                 config.add_user(user.clone(), store_password_in_plaintext)?;
-                Ok(user)
+                user
             } else {
-                Ok(config.get_users()[selection].clone())
+                config.get_users()[selection].clone()
             }
-        }
+        };
+
+        user.current_user = true;
+        config.set_current_user(&user);
```

Comment:
This seems strange - do we still need `user` for anything or can we just do this:

```suggestion
        config.set_current_user(&user);
```